cdap-api/src/main/java/co/cask/cdap/api/lineage/Operation.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+/**
+ * Abstract base class to represent Operation with name and description.
+ */
+public abstract class Operation {
+  private final String name;
+  private final String description;
+
+  protected Operation(String name, String description) {
+    this.name = name;
+    this.description = description;
+  }
+
+  /**
+   * @return the name of the operation, typically its "READ"
```

Comment:
its/it's

Reply:
it is typically a READ? Perhaps say: for example, "read" or "concatenate".

Reply:
is there a limitation on the form of the name? Allowed characters etc.? If so, it should be documented here

Reply:
It represents the name of the Field(https://github.com/caskdata/cdap/blob/develop/cdap-api-common/src/main/java/co/cask/cdap/api/data/schema/Schema.java#L99) for which we do not impose any restriction, however will need to double check it. Let me know if you think we should impose some restrictions.